crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space/metric_exchange.rs

```diff
@@ -138,16 +147,15 @@ impl MetricExchangeSync {
         {
             let mx = out.clone();
             tokio::task::spawn(async move {
-                let mut last_extrap_cov = tokio::time::Instant::now()
-                    .checked_sub(std::time::Duration::from_secs(60 * 60))
-                    .unwrap();
```

Comment:
What?? This panics?? Why? Is it at all reasonable, or a terrible bug? 

We use Instants in other places too, do we need to change every instance of Instant + Duration? Maybe we need some careful documentation to make sure this doesn't happen elsewhere.

Is it Is it the unwrap that panics, or is it checked_sub itself? Could we just use unwrap_or_else to produce a default value which works in this case?

Reply:
It's the underlying operating system type... linux uses something like a i64, which can be subtracted from, macOs uses something like a u64 which cannot be checked_sub-ed. There's nothing we can unwrap_or_else to, because there's no way to create an instant that is before the program start time on macOs.

Reply:
i'm guessing it can sometimes be `0` or something?

Reply:
Yeah, [`u64` can definitely be `checked_sub`'d](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=943e922057a7d9e53bc45b3fee0e4b07), so that doesn't seem like the issue. Somehow an invalid instant is being created? Could it be that this is in some weird CI environment with system clock starting at the UNIX epoch?

Reply:
o_O - I think there's some confusion here. On macOs, the Instant type is backed by something like a u64 that represents a program runtime counter. It starts at zero when the application is run, so if you try to checked_sub with a value greater than the current program runtime, it'll die on the unwrap, because it can't unwrap a None - we can't use a None, we need a valid Instant.

@maackle

Reply:
- https://github.com/rust-lang/rust/blob/master/library/std/src/sys/unix/time.rs#L154
- https://developer.apple.com/documentation/kernel/1462446-mach_absolute_time

Reply:
@neonphog looking at the code, do we even need this subtraction here?

Looks like you're setting the "last time" artificially as an hour before the current moment, just as an initial value to ensure that the loop runs once immediately. Couldn't you just set the time to None to start with, so that the loop always runs the first time, and then to Some(now) on subsequent runs? That would be better in line with the actual meaning of "last run". There is no last run on the first run.

Am I interpreting this correctly? Seems like this would be more straighforward than having to remove the Instant type and revert to a raw i64, seems like an unnecessary step away from using semantically meaningful types..

Reply:
@maackle - that might work for this `last_extrap_cov`, but would be much more awkward for the `last_sync` in the `RemoteRef` above that needs to be initialized to something before the `tick` function call comes in and needs to know if it should be run or not.

Reply:
Oh, sorry, @maackle  I misread your suggestion. Yes I could put both of them in an Option. Can I ask what you dislike about my solution? I prefer not having to match / if branch around options if I can avoid it.

Reply:
I dislike that it replaces `Instant`, which is the proper semantic type, with a primitive asemantic `i64` which is to be interpreted as microseconds, and it pretends that the last sync happened an hour ago when there was no last sync, which isn't true. Small fries at the end of the day, yes, but this PR is kind of saying that Instant is unacceptable to use here, but it basically still is.

If you wanted to do it this way I'd suggest at least using our `Timestamp` type which helps us avoid the millis/micros mismatches we were having. 

Btw you can still get the comparison as a one-liner with: 

```rust
if last_extrap_cov.map(|t| t.elapsed().as_millis() > EXTRAP_COV_CHECK_FREQ_MS).unwrap_or(true)
```

Reply:
@maackle - updated, let me know if this works for you